src/web/user_agent.rs

```diff
@@ -54,9 +54,11 @@ pub struct DeviceInfo {
 }
 
 impl DeviceInfo {
+    /// "Legacy" means that it can only display tiles that are available from
+    /// remote settings. Currently, that's limited to just desktop devices that
+    /// are before v. 91
     pub fn legacy_only(&self) -> bool {
-        self.os_family == OsFamily::IOs && self.ff_version < 36
-            || self.os_family != OsFamily::IOs && self.ff_version < 91
+        self.ff_version < 91
```

Comment:
Sorry to persist on this but I'm hoping to enable non legacy tiles for firefox-ios dev (without them having to tweak their dev version number) which still needs something like this:
```suggestion
        matches!(self.form_factor, FormFactor::Desktop | FormFactor::Other) && self.ff_version < 91
```

I doubt `Other` is really necessary but I'm not sure what produces it and there's hardly any traffic from it so 🤷